src/nvim/eval.c

```diff
@@ -5421,6 +5421,9 @@ static int list_join(garray_T *gap, list_T *l, char_u *sep, int echo_style, int
   garray_T join_ga;
   int retval;
 
+  if (l->lv_len < 1)
+    return OK;
```

Comment:
@bfredl Even if you want `bool` for some reason it is better to add two `#define`s and typedef for bool. Your solution is worse because:

1. Convention adds cognitive load.
2. `if (func())` is less descriptive as `if (func() == OK)`.
3. It is easier to automatically find conventions violation for functions that use enum value then for boolean functions.
4. It makes it easy to violate convention by accident.